temba/api/v2/views.py

```diff
@@ -4410,6 +4414,114 @@ def request__get_separated_names_and_uuids(self, field_name):
                 separated_values["names"].append(value)
         return separated_values
 
+    def name_uuid_filtering(self, queryset, field_name, uuid_key=None, name_key=None, filter_type="filter"):
+        values_fo_filter_by = self.request__get_separated_names_and_uuids(field_name)
+        if values_fo_filter_by["uuids"]:
+            queryset = getattr(queryset, filter_type)(**{f"{uuid_key}__in": values_fo_filter_by["uuids"]})
+        if values_fo_filter_by["names"]:
+            filters_by_name = Q()
+            for filter_value in values_fo_filter_by["names"]:
+                filters_by_name |= Q(**{f"{name_key}__iexact": filter_value})
+            queryset = getattr(queryset, filter_type)(filters_by_name)
+
+        if any(values_fo_filter_by.values()):
+            self.applied_filters[field_name] = dict(item for item in values_fo_filter_by.items() if item[1])
+        return queryset
+
+    def get_contacts(self, count_only=False):
+        org = self.request.user.get_org()
+        contacts = Contact.objects.filter(org=org, status=Contact.STATUS_ACTIVE, is_active=True).distinct()
+
+        # filter by channel
+        contacts = self.name_uuid_filtering(contacts, "channel", "urns__channel__uuid", "urns__channel__name")
```

Comment:
Here we have a joining of two extra tables, that probably can be indexed by elasticsearch.
 `contacts_contact` -> `contacts_contacturn` -> `channels_channel`

Reply:
Yep, can we use the celery for it, having a cronjob to index it overnight daily? I think that could work and would be less time than make the changes on the RP-Indexer for it, I guess.